### ISSUE TITLE
middleware is not used in production with custom server

### ISSUE THREAD

**Issue by Mephiztopheles** [vercel/next.js#69257]
### Link to the code that reproduces this issue

https://github.com/Mephiztopheles/custom-server-middleware

### To Reproduce

1. Start the application (`npm run dev`)
2. Visit localhost:3000 in the browser
3. Click both links and get redirected
4. Stop the server and run the build (`npm run build`)
5. Start the production server (`npm run start`)
6. Visit localhost:3000 again and click both links
7. **Only redirect-2 works** (does not work for me in production but might be a different problem then)

### Current vs. Expected behavior

I expect the middleware to be used in production too.
If i remove NODE_ENV=production, the middleware is used again, but then next is not running in production mode

Also, I expect the redirects in the config to work in my production environment

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #40-Ubuntu SMP PREEMPT_DYNAMIC Fri Jul  5 10:34:03 UTC 2024
  Available memory (MB): 48119
  Available CPU cores: 16
Binaries:
  Node: 22.3.0
  npm: 10.8.1
  Yarn: N/A
  pnpm: 9.4.0
Relevant Packages:
  next: 15.0.0-canary.128 // Latest available version is detected (15.0.0-canary.128).
  eslint-config-next: N/A
  react: 19.0.0-rc-eb3ad065-20240822
  react-dom: 19.0.0-rc-eb3ad065-20240822
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Middleware

### Which stage(s) are affected? (Select all that apply)

next start (local), Other (Deployed)

### Additional context

I have used the website template from payload cms, which may lead to the redirects from the config not beeing used.

But the middleware is also missing in the default setup, what you can see in my example

**Comment by Mephiztopheles** [comment]
Can one at least verify this?!
Its almost two months ago.

No developers seem to care about tickets these days...

**Comment by MohamadrezaKazemian** [comment]
im facing the same issue

**Comment by marcoshernanz** [comment]
Thanks for the reproduction. This is caused by calling the private `next/dist/build` API with a project path that ends in `/` (`path.join(__dirname, "../")`). On the reported version, that produces an empty middleware manifest, so there is no middleware to run in production.

Using the documented `"build": "next build"` script works with the same custom server. Changing the private call to `path.resolve(__dirname, "..")` also restores the middleware manifest, but `next/dist/*` is not a supported public API.

Closing since this is not a custom-server middleware bug. Please use `next build` for production builds.

**Comment by Mephiztopheles** [comment]
Yeah, Im pretty sure I didn't make that code up, so either the website or another framework told me to use this code.

But as of today this issue is irrelevant to me, so I accept this answer. It's two years ago...
